spirv_reflect.c

```diff
@@ -3296,30 +3409,57 @@ static SpvReflectResult ParseStaticallyUsedResources(SpvReflectPrvParser* p_pars
 
   // Do set intersection to find the used uniform and push constants
   size_t used_uniform_count = 0;
-  SpvReflectResult result0 = IntersectSortedAccessedVariable(p_used_accesses, used_acessed_count, uniforms, uniform_count,
-                                                             &p_entry->used_uniforms, &used_uniform_count);
+  result = IntersectSortedAccessedVariable(p_used_accesses, used_acessed_count, uniforms, uniform_count, &p_entry->used_uniforms,
+                                           &used_uniform_count);
+  if (result != SPV_REFLECT_RESULT_SUCCESS) {
+    SafeFree(p_used_accesses);
+    return result;
+  }
 
   size_t used_push_constant_count = 0;
-  SpvReflectResult result1 =
-      IntersectSortedAccessedVariable(p_used_accesses, used_acessed_count, push_constants, push_constant_count,
-                                      &p_entry->used_push_constants, &used_push_constant_count);
+  result = IntersectSortedAccessedVariable(p_used_accesses, used_acessed_count, push_constants, push_constant_count,
+                                           &p_entry->used_push_constants, &used_push_constant_count);
+  if (result != SPV_REFLECT_RESULT_SUCCESS) {
+    SafeFree(p_used_accesses);
+    return result;
+  }
 
   for (uint32_t i = 0; i < p_module->descriptor_binding_count; ++i) {
     SpvReflectDescriptorBinding* p_binding = &p_module->descriptor_bindings[i];
+    uint32_t byte_address_buffer_offset_count = 0;
+
     for (uint32_t j = 0; j < used_acessed_count; j++) {
       if (p_used_accesses[j].variable_ptr == p_binding->spirv_id) {
         p_binding->accessed = 1;
+
+        if (HasByteAddressBufferOffset(p_used_accesses[j].p_node, p_binding)) {
+          byte_address_buffer_offset_count++;
+        }
+      }
+    }
+
+    // only if SPIR-V has ByteAddressBuffer user type
+    if (byte_address_buffer_offset_count > 0) {
+      // possible not all allocated offset slots are used, but this will be a max per binding
+      p_binding->byte_address_buffer_offsets = (uint32_t*)calloc(byte_address_buffer_offset_count, sizeof(uint32_t));
```

Comment:
If there are more than one entrypoint, this reallocs without freeing the previous alloc, then there is a memory stomp in ParseByteAddressBuffer.

Something like this fixes it:

`	if (p_binding->byte_address_buffer_offset_count > 0) {
			uint32_t *prev_byte_address_buffer_offsets = p_binding->byte_address_buffer_offsets;
			p_binding->byte_address_buffer_offsets = (uint32_t *)calloc(byte_address_buffer_offset_count + p_binding->byte_address_buffer_offset_count, sizeof(uint32_t));
			memcpy(p_binding->byte_address_buffer_offsets, prev_byte_address_buffer_offsets, sizeof(uint32_t) * p_binding->byte_address_buffer_offset_count);
			SafeFree(prev_byte_address_buffer_offsets);
		}
		else {
			// possible not all allocated offset slots are used, but this will be a max per binding
			p_binding->byte_address_buffer_offsets = (uint32_t *)calloc(byte_address_buffer_offset_count, sizeof(uint32_t));
		}`